### DISCUSSION TITLE
New rel for history

### DISCUSSION THREAD

**Discussion by HadrienGardeur** [opds-community/drafts#44] [Ideas]
While OPDS already contains `rel` values for:

- a bookshelf: http://opds-spec.org/shelf
- and subscriptions: http://opds-spec.org/subscriptions

... it currently lacks a similar `rel` for historical transactions.

This discussion is a proposal to introduce a new dedicated `rel` value: http://opds-spec.org/history

I expect the main use case for this to be related to public/academic/school libraries, where the returned feed would include publications that have been borrowed before but are no longer included in the user's bookshelf.

Any thoughts/feedback on this? cc @leonardr

**Comment by leonardr** [comment]
A couple thoughts:

We (SimplyE) will most likely implement this using a user profile setting that allows a user to opt-in to history tracking. It would be good to define a key for this setting, either in this spec or in the user profile spec.

A history feed can get quite large and has a lot in common with the Complete Acquisition Feeds described in the OPDS 1.2 spec. We should recommend it be presented in reverse-chronological order.

A history feed can contain the same publication more than once. I don't think this is going to be a problem.

If the publisher of a history feed wants to explain when the transaction happened, they might be able to use the same extensions we'll be using to talk about the durations of loans. Basically, we would implement this by continuing to provide the loan dates after the loan had expired.

**Reply by HadrienGardeur** [reply]
> We (SimplyE) will most likely implement this using a user profile setting that allows a user to opt-in to history tracking. It would be good to define a key for this setting, either in this spec or in the user profile spec.

Do you want to provide this information to display it in the user profile or also to toggle this feature on/off?

> A history feed can get quite large and has a lot in common with the Complete Acquisition Feeds described in the OPDS 1.2 spec. We should recommend it be presented in reverse-chronological order.

Agree about reverse-chronological order as a best practice, but I'm not so sure that we have that much in common with Complete Acquisition Feeds (a concept that doesn't exist in OPDS 2.0). Aside from an order, we should also recommend pagination as a best practice.

> If the publisher of a history feed wants to explain when the transaction happened, they might be able to use the same extensions we'll be using to talk about the durations of loans. Basically, we would implement this by continuing to provide the loan dates after the loan had expired.

How would that work? Using `since` and `until` in the acquisition link would be confusing IMO.
